apollo-router/src/query_planner/fetch.rs

```diff
@@ -266,10 +266,30 @@ impl FetchNode {
             // when errors have been redacted in the include_subgraph_errors module.
             // Unfortunately, not easy to fix here, because at this point we don't
             // know if we should be redacting errors for this subgraph...
-            .map_err(|e| FetchError::SubrequestHttpError {
-                status_code: None,
-                service: service_name.to_string(),
-                reason: e.to_string(),
+            .map_err(|e| {
+                if !e.is::<FetchError>() {
+                    FetchError::SubrequestHttpError {
+                        status_code: None,
+                        service: service_name.to_string(),
+                        reason: e.to_string(),
+                    }
+                } else {
+                    let inner = e.downcast::<FetchError>().unwrap();
+                    match *inner {
+                        FetchError::SubrequestHttpError { status_code, .. } => {
+                            FetchError::SubrequestHttpError {
+                                status_code,
+                                service: service_name.to_string(),
+                                reason: inner.to_string(),
+                            }
+                        }
+                        _ => FetchError::SubrequestHttpError {
+                            status_code: None,
+                            service: service_name.to_string(),
+                            reason: inner.to_string(),
+                        },
+                    }
+                }
```

Comment:
I think it's the same than just doing (but maybe I'm missing something):
```rust
FetchError::SubrequestHttpError {
  status_code,
  service: service_name.to_string(),
  reason: inner.to_string(),
}
```

Reply:
With the match case on the downcast from your previous comment, I was able to simplify even further, removing the `is` check and simply pushing forward the existing `FetchError::SubrequestHttpError` (WITH `status_code` already included) if we find a match. Thanks for the feedback! This removes a copy as well from the previous implementation 💯